### ISSUE TITLE
StringTransport.get{Host,Peer} sometimes return a tuple instead of an IAddress

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4401]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4401|
|Type|defect|
|Created|2010-04-15 13:17:21Z|
|Branch|https://github.com/twisted/twisted/tree/stringtransport-endpoint-addresses-4401|

These methods should be updated to always return some kind of `IAddress`.  Probably an `IPv4Address` by default (it's already possible to pass in whatever specific address you want returned here, so using ipv4 as a default doesn't limit any functionality).

I noticed this while looking at [#4394](https://github.com/twisted/twisted/issues/4394).  The ident tests should have been emitting deprecation warnings, but weren't, because they use `StringTransport` instead of an actual TCP connection.

<details><summary>Searchable metadata</summary>

```
trac-id__4401 4401
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_stringtransport_endpoint_addresses_4401 branches/stringtransport-endpoint-addresses-4401
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1271337441000000 1271337441000000
changetime__1272471508000000 1272471508000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [28872]) Branching to 'stringtransport-endpoint-addresses-4401'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [28873]) Return IPv4Address instances from getHost and getPeer

refs [#4401](https://github.com/twisted/twisted/issues/4401)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

I'm a bit bothered by the potential for breaking people's tests. OTOH, depending on the old behavior was seriously borked, and this shouldn't break any production code.

So, merge it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [28878]) Merge stringtransport-endpoint-addresses-4401

Author: exarkun
Reviewer: itamar
Fixes: [#4401](https://github.com/twisted/twisted/issues/4401)

Change `StringTransport.getHost` and `StringTransport.getPeer`
to return `IPv4Address` instances by default, rather than
old-style two-tuples of address information.  This may cause
new deprecation warnings to be emitted in tests which are
using the deprecated indexing API of `IPv4Address`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
